### PR TITLE
Deprecate the global DOMPurify

### PR DESCRIPTION
**Pull request by ChristophWurst**
Found at https://github.com/nextcloud/server/pull/17123.

We don't really use it globally anymore, so no reason to keep it exported.

```
$ rg dompurify -M 300 -t js
core/src/globals.js
78:import DOMPurify from 'dompurify'

settings/js/vue-5.js
1:[Omitted long line with 1 matches]

core/js/dist/main.js
338:[Omitted long line with 1 matches]

settings/src/components/appDetails.vue
103:import dompurify from 'dompurify'
193:			return dompurify.sanitize(
```

### CODE REVIEW

File: core/src/globals.js [hunk 1]
```diff
@@ -107,8 +107,8 @@ window['Clipboard'] = ClipboardJS
 window['ClipboardJS'] = ClipboardJS
 window['cssVars'] = cssVars
 window['dav'] = dav
-window['DOMPurify'] = DOMPurify
-window['Handlebars'] = Handlebars
+setDeprecatedProp('DOMPurify', () => DOMPurify, 'The global DOMPurify is deprecated, ship your own')
+window['Handlebars'] = HandlebarsS
```
Comment (georgehrke) [inline core/src/globals.js]:
```suggestion
window['Handlebars'] = Handlebars
```

Comment (ChristophWurst) [reply core/src/globals.js]: dang it!

**Review by rullzer — APPROVED** [review]

**Review by blizzz — APPROVED** [review]

**Comment by blizzz** [comment]
conflicts

**Pull request merged**